src/rgw/rgw_rados.cc

```diff
@@ -12345,3 +12355,23 @@ int RGWRados::delete_obj_aio(rgw_obj& obj, rgw_bucket& bucket,
   return ret;
 }
 
+int rgw_compression_info_from_attrset(map<string, bufferlist>& attrs, bool& need_decompress, RGWCompressionInfo& cs_info) {
+  if (attrs.find(RGW_ATTR_COMPRESSION) != attrs.end()) {
+    bufferlist::iterator bliter = attrs[RGW_ATTR_COMPRESSION].begin();
```

Comment:
`attrs[RGW_ATTR_COMPRESSION]` is doing another search on `attrs`. consider using the iterator from `find()` instead:
```c++
  auto i = attrs.find(RGW_ATTR_COMPRESSION);
  if (i != attrs.end()) {
    bufferlist::iterator bliter = i->second.begin();
```